lib/beaker/hypervisor/aws_sdk.rb

```diff
@@ -688,18 +688,21 @@ def public_key
       keys = Array(@options[:ssh][:keys])
       keys << '~/.ssh/id_rsa'
       keys << '~/.ssh/id_dsa'
-      key_file = nil
-      keys.each do |key|
-        key_filename = File.expand_path(key + '.pub')
-        key_file = key_filename if File.exists?(key_filename)
+      key_file = keys.find do |key|
+        key_pub = key + '.pub'
+        if File.exist?(File.expand_path(key_pub)) && File.exist?(File.expand_path(key))
+          true
```

Comment:
@hunner, you can just use this if statement as a standalone. You don't need to return `true` or `false`.